src/settings.js

```diff
@@ -50,7 +50,7 @@ export class SettingsStore {
    * @param {Object | undefined} config the configuration
    */
   constructor(config) {
-    this.#client = new DynamoDBClient(config);
+    this.#client = config.client || new DynamoDBClient(config);
```

Comment:
is this added for unit tests?

Reply:
Yes, its so I can provide in a mocked client for the unit tests.